internal/hcl/modules/fetch.go

```diff
@@ -136,20 +183,13 @@ func (p *PackageFetcher) fetchFromRemote(moduleAddr, dest string) (bool, error)
 	// I'm not sure if we really need it, but added it just in case/
 	decompressors["tar.tbz2"] = new(getter.TarBzip2Decompressor)
 
-	getters := make(map[string]getter.Getter, len(getter.Getters))
-	for k, g := range getter.Getters {
-		getters[k] = g
-	}
-
-	getters["git"] = &CustomGitGetter{new(getter.GitGetter)}
-
 	client := getter.Client{
 		Src:           moduleAddr,
 		Dst:           dest,
 		Pwd:           dest,
-		Mode:          getter.ClientModeDir,
+		Mode:          getter.ClientModeAny,
```

Comment:
Terragrunt needs this and it seems to work okay. Go-getter still detects archives as directories.